### DISCUSSION TITLE
Maintaining a common devcontainer across many repositories

### DISCUSSION THREAD

**Discussion by TWolversonReply** [devcontainers/community#81] [Q&A]
Suppose I work in an organisation with many microservices in their own repos. Most or all of these microservices share an implementation technology so by and large they will have the same tool needs. I expect to pepper various new technology options across a random selection of these repositories according to need, so generally if we support a new tool, we would always want to add it to our standard devcontainer. I don't want to have to pull request a devcontainer update to every single repo every time I make a change. Is there a recommended pattern for maintaining a devcontainer definition in one place and automatically synchronising it with many small repositories?

**Comment by bamurtaugh** [comment]
Thanks for opening!

One idea: You could prebuild an image (https://containers.dev/guide/prebuild), and have each of the smaller repos reference that prebuilt image. If they can all use the `latest` tag, then on future rebuilds, the smaller repos could automatically pull the changes that went to the latest version of the prebuild.

**Reply by TWolversonReply** [reply]
That looks like that would work! Should be a better option than my prior set of ideas which was nothing. I guess that would take a few more steps to try things out compared to just adding a block to the devcontainer.json, which might hamper devs who may not be experts. Overall that seems like less work than duplicating changes though.

**Reply by bamurtaugh** [reply]
Sounds great! And if you have feedback as you try out prebuilds, please let us know!

**Reply by kerryhatcher** [reply]
That is exactly the pattern I was considering before stumbling across this. Thanks for the affirmation.
